--- a/os_dep/linux/ioctl_cfg80211.c
+++ b/os_dep/linux/ioctl_cfg80211.c
@@ -466,8 +466,6 @@ u8 rtw_cfg80211_ch_switch_notify(_adapte
 		cfg80211_ch_switch_started_notify(adapter->pnetdev, &chdef, 0, 0, false);
 #elif (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 11, 0)) || (RHEL_RELEASE_CODE >= RHEL_RELEASE_VERSION(8,0))
 		cfg80211_ch_switch_started_notify(adapter->pnetdev, &chdef, 0, false);
-#else
-		cfg80211_ch_switch_started_notify(adapter->pnetdev, &chdef, 0);
 #endif
 		goto exit;
 	}
--- a/os_dep/linux/os_intfs.c
+++ b/os_dep/linux/os_intfs.c
@@ -2794,7 +2798,13 @@ static int netdev_vir_if_close(struct ne
 
 #ifdef CONFIG_IOCTL_CFG80211
 	wdev->iftype = NL80211_IFTYPE_MONITOR;
+	#if (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 19, 2)) || defined(BUILD_OPENWRT)
+	wdev->links[0].client.current_bss = NULL;
+	#elif (LINUX_VERSION_CODE >= KERNEL_VERSION(5, 19, 2))
+	wdev->connected = NULL;
+	#else
 	wdev->current_bss = NULL;
+	#endif
 	rtw_scan_abort(padapter);
 	rtw_cfg80211_wait_scan_req_empty(padapter, 200);
 	adapter_wdev_data(padapter)->bandroid_scan = _FALSE;
